src/analysis/special_functions/trigonometric.lean

```diff
@@ -2744,7 +2744,7 @@ lemma cos_surjective : function.surjective cos :=
 begin
   intro x,
   obtain ⟨w, w₀, hw⟩ : ∃ w ≠ 0, 1 * w * w + (-2 * x) * w + 1 = 0,
-  { rcases exists_quadratic_eq_zero one_ne_zero (exists_eq_mul_self _) with ⟨w, hw⟩,
+  { rcases exists_quadratic_eq_zero (@one_ne_zero ℂ _ _) (exists_eq_mul_self _) with ⟨w, hw⟩,
```

Comment:
For reasons I only partially understand, the new instances added in `algebra/ring/basic` mean we need to give the elaborator a little more help here.

Reply:
I'd like to understand this better too, if any more experienced members have any hints.